internal/toolexec/proxy/flags.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package proxy
+
+import (
+	"reflect"
+	"strings"
+)
+
+const structTagKey = "ddflag"
+
+// parseFlags walks through the given arguments and sets the flagSet values
+// present in the argument list. Unknown options, not present in the flagSet
+// are accepted and skipped. The argument list is not modified.
+func parseFlags(flagSet interface{}, args []string) {
+	flagSetValueMap := makeFlagSetValueMap(flagSet)
+
+	i := 0
+	for i < len(args)-1 {
+		_, shift := parseOption(flagSetValueMap, args[i], args[i+1])
+		i += shift
+	}
+
+	if i < len(args) {
+		_, _ = parseOption(flagSetValueMap, args[i], "")
+	}
+}
+
+func makeFlagSetValueMap(flagSet interface{}) map[string]reflect.Value {
+	v := reflect.ValueOf(flagSet).Elem()
+	typ := v.Type()
+	flagSetValueMap := make(map[string]reflect.Value, v.NumField())
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+		if tag, ok := field.Tag.Lookup(structTagKey); ok {
+			flagSetValueMap[tag] = v.Field(i)
+		}
+	}
+	return flagSetValueMap
+}
+
+// parseOption parses the given current argument and following one according to
+// the go Flags syntax.
+func parseOption(flagSetValueMap map[string]reflect.Value, arg, nextArg string) (nonOpt bool, shift int) {
+	if arg[0] != '-' {
+		// Not an option, return the value and shift by one.
+		return true, 1
+	}
+
+	// Split the argument by its first `=` character if any, and check the
+	// syntax being used.
+	kv := strings.SplitN(arg, "=", 2)
+	option := kv[0]
+
+	flag, exists := flagSetValueMap[option]
+
+	if len(kv) == 2 {
+		// `-opt=val` syntax
+		value := kv[1]
```

Comment:
The `strings.Cut` function is a better fit for this use-case:

```suggestion
	option, value, hasValue := strings.Cut(arg, "=")
	flag, exists := flagSetValueMap[option]
	
	if hasValue {
		// `-opt=val` syntax
```